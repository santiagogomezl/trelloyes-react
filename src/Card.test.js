import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card.js';
import renderer from 'react-test-renderer';

describe('My component', () => {
    // all my test cases here. Test Suite

    //Smoke Test
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
    });


    //Screenshot Test
    it('renders the UI as expected', () => {
        const tree = renderer
        .create(<Card title={'Card Title'} content={'Card Content Lorem Ipsum'}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();  
        });

})