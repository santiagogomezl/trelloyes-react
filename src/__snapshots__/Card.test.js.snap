// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    Card Title
  </h3>
  <p>
    Card Content Lorem Ipsum
  </p>
</div>
`;
